From 01448b1f8574160858ca25f592d83138dede5fc3 Mon Sep 17 00:00:00 2001
From: Jason Liu <jason.hui.liu@nxp.com>
Date: Thu, 25 Oct 2018 10:32:08 +0800
Subject: [PATCH 4921/5242] MLK-19989 drivers: mxc: hdp-cec: make the hdp-cec
 kernel thread freezeable

commit  50acd9fc7443e7a4ef211c40e67582ec7d752794 from
https://source.codeaurora.org/external/imx/linux-imx.git

the hdp-cec kernel thread should be freezeable during system suspend phase,
othwerwise, it will cause issues(hang) since it will still try to access
some resources such as clocks which are off during system suspend process.

Signed-off-by: Jason Liu <jason.hui.liu@nxp.com>
Cc: Sandor Yu <sandor.yu@nxp.com>
Cc: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Acked-by: Sandor Yu <sandor.yu@nxp.com>
Reviewed-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp-cec/imx-hdp-cec.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/hdp-cec/imx-hdp-cec.c b/drivers/mxc/hdp-cec/imx-hdp-cec.c
index e1aea1f..068ae8e 100644
--- a/drivers/mxc/hdp-cec/imx-hdp-cec.c
+++ b/drivers/mxc/hdp-cec/imx-hdp-cec.c
@@ -22,6 +22,7 @@
 #include <linux/version.h>
 #include <linux/workqueue.h>
 #include <linux/kthread.h>
+#include <linux/freezer.h>
 #include <media/cec.h>
 #include <soc/imx8/soc.h>
 
@@ -173,8 +174,10 @@ static int cec_poll_worker(void *_cec)
 	int num_rx_msgs, i;
 	int sts;
 
+	set_freezable();
+
 	for (;;) {
-		if (kthread_should_stop())
+		if (kthread_freezable_should_stop(NULL))
 			break;
 
 		/* Check TX State */
-- 
1.7.9.5

